### ISSUE TITLE
Documentation gap: not clear how to add a responder in the action for an incident workflow

### ISSUE THREAD

**Issue by james00002** [PagerDuty/terraform-provider-pagerduty#727]
### Affected Resource(s)
- pagerduty_incident_workflow

### Documentation Gap
The documentation ([here](https://registry.terraform.io/providers/PagerDuty/pagerduty/latest/docs/resources/incident_workflow#example-usage)) provides an example of how to send a status update, but examples for other actions are needed.

For example, if action = "pagerduty.com:incident-workflows:add-responders:2" it is not clear what value should go in the input for "Responders"

Even checking the list-actions api ([here](https://developer.pagerduty.com/api-reference/aa192a25fac39-list-actions)) does not help, it just says:
        {
          "name": "Responders",
          "description": "The specific Responders to be added",
          "type": "json",
          "is_required": true,
          "is_hidden": false,
          "advanced": false,
          "metadata": "{\"ref\":{\"type\":\"responder\"}}"
        },

But that still gives no guidance about what the actual value should be in terraform.
The most obvious guess of using `data.pagerduty_user.some_user.id` does not work.

**Comment by james00002** [comment]
By messing around with the API, I was able to discover that the expected value is a string containing  an escaped json list, in a format like this, with users having type "user" and teams having type "escalation_policy"

`"[{\"id\":\"PMXXXXX\",\"type\":\"escalation_policy\"},{\"id\":\"PNYYYYY\",\"type\":\"user\"}]"`

Which is perfectly logical, but I would never had guess it without mucking about in the API and guessing-and-checking. the docs could definitely use an example.

**Comment by james00002** [comment]
I can confirm that this works pretty well: 

      value = jsonencode([{"id":data.pagerduty_user.some_user.id, "type": "user"}])

or 

      value = jsonencode([{"id":data.pagerduty_team.some_team.id, "type": "escalation_policy"}])

**Comment by imjaroiswebdev** [comment]
You are right, We need to improve documentation with examples in this matter; same happens with **Add Stakeholders** action.

However just to complement your solution, it's good to know that you may also use [JSON heredoc](https://developer.hashicorp.com/terraform/language/expressions/strings#generating-json-or-yaml) to set action input value, and also *HCL* syntax directly into `jsonencode` Terraform function, as follow...

```hcl
resource "pagerduty_incident_workflow" "my_first_workflow" {
  name        = "Example Incident Workflow"
  description = "This Incident Workflow is an example"
  step {
    name   = "Add Responders"
    action = "pagerduty.com:incident-workflows:add-responders:2"
    input {
      name = "Responders"
      # value = <<JSON # Valid, but discouraged
      # [
      #   {
      #     "id": "${pagerduty_escalation_policy.tf_escalation_local1.id}",
      #     "type": "escalation_policy"
      #   },
      #   {
      #     "id": "${pagerduty_user.tf_user_local1.id}",
      #     "type": "user"
      #   }
      # ]
# JSON
      value = jsonencode([
        {
          id   = pagerduty_escalation_policy.tf_escalation_local1.id
          type = "escalation_policy"
        },
        {
          id   = pagerduty_user.tf_user_local1.id
          type = "user"
        },
      ])
    }
  }
}

```

Although, using *JSON heredoc* is a discouraged practice for Terraform code.

On top of that, this would be a good opportunity to also document directly in the TF resource reference which are the supported objects to set as Respoders and Stakeholders, despite the fact this is already documented in the [Product's docs](https://support.pagerduty.com/docs/add-responders#add-responders-to-an-incident).

So, I'll keep this issue open to track the documentation PR origin.

Thank you @james00002 for sharing your solution to this and setting the precedent for documentation improvement in the Provider.
